raft.go

```diff
@@ -250,6 +314,24 @@ func NewRaft(conf *Config, fsm FSM, logs LogStore, stable StableStore, snaps Sna
 		return nil, err
 	}
 
+	// Scan through the log for any configuration change entries
+	snapshotIndex, _ := r.getLastSnapshot()
+	for index := snapshotIndex + 1; index <= lastLog.Index; index++ {
+		var entry Log
+		if err := r.logs.GetLog(index, &entry); err != nil {
+			r.logger.Printf("[ERR] raft: Failed to get log at %d: %v", index, err)
+			panic(err)
+		}
+		// TODO: support LogAddPeer, LogRemovePeer too
+		if entry.Type == LogConfiguration {
+			r.configurations.committed = r.configurations.latest
```

Comment:
Don't think so. Not everything we find in the log will be committed. We do know a new configuration entry can't be created until the prior one was committed => the penultimate configuration in the log is committed. The last one may not be.

Note that latest is used almost everywhere. If latest is in fact committed, this server will never be asked to fall back to the previous one, and it should all work out just fine.